src/Rfc1035StubDnsResolver.php

```diff
@@ -218,14 +218,16 @@ public function resolve(string $name, ?int $typeRestriction = null, ?Cancellatio
      * Reloads the configuration in the background.
      *
      * Once it's finished, the configuration will be used for new requests.
+     *
+     * May return null if the configuration cannot be loaded.
      */
-    public function reloadConfig(): DnsConfig
+    public function reloadConfig(): ?DnsConfig
     {
         if ($this->pendingConfig) {
             return $this->pendingConfig->await();
         }
 
-        $this->pendingConfig = async(function (): DnsConfig {
+        $this->pendingConfig = async(function (): ?DnsConfig {
```

Comment:
Changing the return type here is a BC break. :-(

Reply:
Yeah, but the current logic is borked

Reply:
We could return an "empty" DnsConfig with no nameservers to avoid the BC break. We'd still need to set the config within the object to null.